lib/bot-service.js

```diff
@@ -70,9 +79,11 @@ function decodeToken (token) {
         if (decoded.hasOwnProperty('bot')) {
             return decoded.bot;
         }
-        logger.error('Invalid Response. Missing key `bot` in response JSON');
+      logger.error('Invalid Response. Missing key `bot` in response JSON');
+      return new Error('Invalid Response. Missing key `bot` in response JSON');
     } catch (error) {
-        logger.error('Unable to decode response from BoT server.');
+      logger.error('Unable to decode response from BoT server.');
+      return new Error('Unable to decode response from BoT server.')
```

Comment:
`;`

Reply:
why are you returning an error
better throw it

Reply:
I'm returning an error because the function call is synchronous and expects a return.